### DISCUSSION TITLE
Plotting a rotated crystal goemetry

### DISCUSSION THREAD

**Discussion by JulianMeck** [mtex-toolbox/mtex#1621] [Ask Anything]
Hi Mtexers,

I want to plot a figure with lattice planes and axes of a single crystal rotated by some set of Euler angles. I am sure this is possible in MTex I just cannot find in the documentation how to do it.

Thanks in advance
Julian

**Comment by lfgmorales** [comment]
Hey Julian,



Would the "rotationbyEuler" would do what you want?


like here


https://mtex-toolbox.github.io/RotationDefinition.html


https://mtex-toolbox.github.io/RotationOperations.html


cheers



Luiz





Luiz F. G. Morales
ScopeM & Structural Geology and Tectonics Group
ETH Zürich
HPM C57.3

Otto-Stern-Weg 3

8093 Zürich

Switzerland

Phone: + 41 44 633 3746

<tel:+41446333746>E-mail: ***@***.***



"No trees were destroyed in the sending of this message, however, a significant number of electrons were terribly inconvenienced"


________________________________
From: JulianMeck ***@***.***>
Sent: 01 November 2022 16:27:43
To: mtex-toolbox/mtex
Cc: Subscribed
Subject: [mtex-toolbox/mtex] Plotting a rotated crystal goemetry (Discussion #1621)


Hi Mtexers,

I want to plot a figure with lattice planes and axes of a single crystal rotated by some set of Euler angles. I am sure this is possible in MTex I just cannot find in the documentation how to do it.

Thanks in advance
Julian

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1621>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN66A3KN3GL2B2CJLV3WGEZG7ANCNFSM6AAAAAARUFJH3I>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1621", "url": "https://github.com/mtex-toolbox/mtex/discussions/1621", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by JulianMeck** [reply]
Hi Luiz,

Thanks for reply. But how I apply that rotation when plotting figure is what I am struggling with. Below is the code I have written so far.

Cheers
Julian


%% load cif file

cs = loadCIF('forsterite');

%% define miller

d1=Miller(1,0,0,cs,'uvw');
d2=Miller(0,1,0,cs,'uvw');
d3=Miller(0,0,1,cs,'uvw');
m1=Miller(1,3,0,cs,'hkl');
m2=Miller(1,3,1,cs,'hkl');
m3=Miller(1,1,2,cs,'hkl');
rot = rotation.byEuler(30*degree,50*degree,10*degree);


%% Plot figure
figure(1)
clf
plot(d1,'upper','labeled')
hold on
plot(d2,'upper','labeled')
plot(d3,'upper','labeled')
plot(m1,'plane','linecolor','r','linewidth',2,'add2all')
plot(m2,'plane','linecolor','g','linewidth',2,'add2all')
plot(m3,'plane','linecolor','b','linewidth',2,'add2all')
hold off

**Comment by zmichels** [comment]
Try:

plot(rot*d1,'upper','labeled')

**Comment by JulianMeck** [comment]
I have tried that but the labels change to be relative to geographic coordinates.
Cheers
Julian

**Comment by zmichels** [comment]
plot(rot*d1,'upper','label',sprintf('[%i%i%i]', d1.u,d1.v,d1.w))
plot(rot*d2,'upper','label',sprintf('[%i%i%i]', d2.u,d2.v,d2.w))
plot(rot*d3,'upper','label',sprintf('[%i%i%i]', d3.u,d3.v,d3.w))

**Reply by JulianMeck** [reply]
Thanks that is great
